### PR TITLE
ASSETS-88904 : Update Asset Thumbnail for Campaign List Entries

### PR DESCRIPTION
**Pull request by TyroneAEM**
Please always provide the [JIRA issue(s)](https://jira.corp.adobe.com/) your PR is for, as well as a description of your changes:

Workfront Task : https://experience.adobe.com/#/@wfadoberm/so:adoberm-Production/workfront/task/663cf46b002751876d8e92b9a31b048f/overview

Test URLs:
https://assets-88904--adobe-gmo--hlxsites.hlx.page/qa/marketing-dashboard

Screenshot 2024-05-14 at 4 00 26 PM

Authoring program-details.docx page

Added the following metadata
listPage : marketing-dashboard

Screenshot 2024-05-15 at 3 31 46 PM

### CODE REVIEW

**Comment by shiv-gup** [comment]
@TyroneAEM Please update the thumbnail on Campaign Detail(Overview) page as well. Same thumbnail needs to be shown on detail page as well.

File: blocks/gmo-campaign-list/AssetsDatasource.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+ * @param {string} - Program Name.
+ * @param {string} - Campaign Name.
+ * @returns {Image <object>} Resolves with image object.
+ * @throws {Error} If an HTTP error or network error occurs.
+ */
+
+export async function searchAsset(programName, campaignName) {
+  const adminConfig = await getAdminConfig();
+  const deliveryURL = await initDeliveryEnvironment();
+  const searchURL = `${deliveryURL}/adobe/assets/search`
```
Comment (shiv-gup) [inline blocks/gmo-campaign-list/AssetsDatasource.js:49]: use `createSearchEndpoint` function of `scripts.js` to get the search endpoint url

File: blocks/gmo-campaign-list/gmo-campaign-list.js [hunk 2]
```diff
@@ -89,7 +95,8 @@ export default async function decorate(block, numPerPage = currentNumberPerPage,
     currentPageInfo.itemCount = campaigns.length;
 
     const listHeaders = buildListHeaders(headerConfig);
-    const listItems = buildCampaignList(campaigns, numPerPage);
+    //const listItems = buildCampaignList(campaigns, numPerPage);
```
Comment (shiv-gup) [inline blocks/gmo-campaign-list/gmo-campaign-list.js:98]: remove this commented code

File: scripts/assets.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+ * @param {string} - Program Name.
+ * @param {string} - Campaign Name.
+ * @returns {Image <object>} Resolves with image object.
+ * @throws {Error} If an HTTP error or network error occurs.
+ */
+
+export async function searchAsset(programName, campaignName, imageWidth = 80) {
+  const adminConfig = await getAdminConfig();
+  const deliveryURL = await initDeliveryEnvironment();
+  const searchURL = `${deliveryURL}/adobe/assets/search`
```
Comment (shiv-gup) [inline scripts/assets.js:49]: use `createSearchEndpoint` function to get the search URL from `script.js` file

File: scripts/assets.js [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+  getSearchIndex,
+  initDeliveryEnvironment,
+  getOptimizedDeliveryUrl
+} from './polaris.js';
+
+import { getAdminConfig } from './site-config.js';
+
+import { logError } from './scripts.js';
+
+async function getRequestHeadersSearchAssets() {
```
Comment (shiv-gup) [inline scripts/assets.js:16]: Can you check if `getSearchClient` from `scripts.js` can be reused here?

Comment (TyroneAEM) [reply scripts/assets.js:16]: It does not look like you can

**Review by shiv-gup — APPROVED** [review]

**Pull request merged**